src/sc2laddercore/LadderConfig.cpp

```diff
@@ -1,76 +1,67 @@
+#include "LadderConfig.h"
+
+#include "Types.h"
+
+#include "ostreamwrapper.h"
+#include "writer.h"
+#include "prettywriter.h"
+
 #include <string>
 #include <map>
 #include <iostream>
 #include <sstream>
-#include <fstream>      
-#include "LadderConfig.h"
+#include <fstream>    
 
-LadderConfig::LadderConfig(const std::string &InConfigFile)
-	:ConfigFileLocation(InConfigFile)
+
+LadderConfig::LadderConfig(const std::string &ConfigFile)
+	:ConfigFileLocation(ConfigFile)
 {
+	// Doing this allows us to write values to the doc
+	// straight off instead of loading from file
+	this->doc.SetObject();
 }
 
 bool LadderConfig::ParseConfig()
 {
-	std::ifstream ifs(ConfigFileLocation, std::ifstream::binary);
-	if (!ifs)
-	{
-		return false;
-	}
-	
-
-	std::string line;
-	while (std::getline(ifs, line))
+	std::ifstream t(this->ConfigFileLocation);
+	if (t.good())
 	{
-		std::istringstream is_line(line);
-		std::string key;
-		if (std::getline(is_line, key, '='))
-		{
-			std::string value;
-			if (std::getline(is_line, value))
-			{
-				TrimString(key);
-				TrimString(value);
-				options.insert(std::make_pair(key, value));
-			}
-		}
+		std::stringstream buffer;
+		buffer << t.rdbuf();
+		std::string ConfigString = buffer.str();
+		return !doc.Parse(ConfigString.c_str()).HasParseError();
 	}
-	return true;
+	else
```

Comment:
This is just the way I'm used to because I understand it easier - it tells me there should be a `return` somewhere in the `if` statement when I see the `else return`.
Is your suggestion a more standard way of doing it?
Regardless, not too fussed which way it is - just asking for my own edification.

Reply:
Nope, this is a matter of taste.

Reply:
Thank you!